plugwise/smile.py

```diff
@@ -519,20 +525,32 @@ async def set_schedule_state(self, loc_id, name, state):
 
         return True
 
-    async def set_preset(self, loc_id, preset):
+    async def _set_preset_legacy(self, preset) -> bool:
+        """Set the given Preset on the relevant Thermostat - from DOMAIN_OBJECTS."""
+        locator: str = f'rule/directives/when/then[@icon="{preset}"].../.../...'
+        if (rule := self._domain_objects.find(locator)) is None:
+            return False
+
+        uri: str = f"{RULES}"
```

Comment:
Needs no typing, needs no string interpolation

```suggestion
        uri = RULES
```